Runtime/Tracking/Follow/Modifier/Property/Rotation/TransformPositionDifferenceRotation.cs

```diff
@@ -0,0 +1,98 @@
+        /// </summary>
+        public Vector3 AngularVelocity { get; protected set; }
+        /// <summary>
+        /// The previous source world position.
+        /// </summary>
+        protected Vector3? previousSourcePosition;
+
+        /// <summary>
+        /// Resets the state of the source previous position.
+        /// </summary>
+        public virtual void ResetPreviousState()
+        {
+            previousSourcePosition = null;
+        }
+
+        /// <summary>
+        /// Rotates the target based on the position difference of the source.
+        /// </summary>
+        /// <param name="source">The source to utilize in the modification.</param>
+        /// <param name="target">The target to modify.</param>
+        /// <param name="offset">The offset of the target against the source when modifying.</param>
+        protected override void DoModify(GameObject source, GameObject target, GameObject offset = null)
+        {
+            Vector3 sourcePosition = source.transform.position;
+            Vector3 targetPosition = target.transform.position;
+
+            if (previousSourcePosition == null)
+            {
+                previousSourcePosition = sourcePosition;
+            }
+
+            float xDegree = FollowOnAxis.xState ? CalculateAngle(target.transform.right, targetPosition, (Vector3)previousSourcePosition, sourcePosition) : 0f;
+            float yDegree = FollowOnAxis.yState ? CalculateAngle(target.transform.up, targetPosition, (Vector3)previousSourcePosition, sourcePosition) : 0f;
+            float zDegree = FollowOnAxis.zState ? CalculateAngle(target.transform.forward, targetPosition, (Vector3)previousSourcePosition, sourcePosition) : 0f;
+
+            AngularVelocity = new Vector3(xDegree, yDegree, zDegree) * (1f / AngularDrag);
+
+            target.transform.localRotation *= Quaternion.Euler(AngularVelocity);
+            previousSourcePosition = sourcePosition;
+        }
+
+        protected virtual void OnDisable()
+        {
+            ResetPreviousState();
+        }
+
+        /// <summary>
+        /// Calculates the rotational angle for an axis based on the difference between two points around the origin.
+        /// </summary>
+        /// <param name="originDirection">The direction representing the axis.</param>
+        /// <param name="originPoint">The angle centre.</param>
+        /// <param name="pointA">The first point to calculate the angle from.</param>
+        /// <param name="pointB">The second point to calculate the angle to.</param>
+        /// <returns>The angle in degrees between the two points.</returns>
+        protected virtual float CalculateAngle(Vector3 originDirection, Vector3 originPoint, Vector3 pointA, Vector3 pointB)
+        {
+            Vector3 heading = pointB - originPoint;
+            Vector3 sideA = pointA - originPoint;
+            if (heading.magnitude.ApproxEquals(0f))
```

Comment:
Save the result of `heading.magnitude` into a local variable and reuse it everywhere in this method - this property is calculated on the fly and does some math we don't want to be done multiple times.